backend/app/api/endpoints/chapters.py

```
@@ -0,0 +1,592 @@
+    except Exception:
+        logger.error("Failed to create chapter", exc_info=True)
+        raise HTTPException(status_code=500, detail="Failed to create chapter")
+
+
+# NOTE: literal sub-paths (/chapters/metadata, /chapters/tab-state) must be
+# registered BEFORE the parameterized /chapters/{chapter_id} route, otherwise
+# FastAPI matches them as chapter_id="metadata"/"tab-state" and they 404.
+@router.get("/{book_id}/chapters/metadata", response_model=ChapterMetadataResponse)
+async def get_chapters_metadata(
+    book_id: str,
+    current_user: Dict = Depends(get_current_user_from_session),
```

Ruff (0.15.20)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)